frontend/src/State.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+const StateContext = React.createContext({});
+
+const StateProvider = (props) => {
```

Comment:
This is an interesting and useful way to engage the React Context API, since it provides access to a setState-like functionality that has closure over the useState's setter functon.